code/game/objects/items/weapons/implants/implant_explosive.dm

```diff
@@ -125,4 +125,49 @@
 
 /obj/item/weapon/implanter/explosive_macro/New()
 	imp = new /obj/item/weapon/implant/explosive/macro(src)
+	..()
+
+
+// Dust implant, for CC officers. Prevents gear theft if they die.
+
+/obj/item/weapon/implant/dust
+	name = "duster implant"
+	desc = "An alarm which monitors host vital signs, transmitting a radio message and dusting the corpse on death."
+
+/obj/item/weapon/implant/dust/get_data()
+	var/dat = {"<b>Implant Specifications:</b><BR>
+				<b>Name:</b> Ultraviolet Corp XX-13 Security Implant<BR>
+				<b>Life:</b> Activates upon death.<BR>
+				<b>Important Notes:</b> Vaporizes organic matter<BR>
+				<HR>
+				<b>Implant Details:</b><BR>
+				<b>Function:</b> Contains a compact, electrically activated heat source that turns its host to ash upon activation, or their death. <BR>
+				<b>Special Features:</b> Vaporizes<BR>
+				"}
+	return dat
+
+/obj/item/weapon/implant/dust/trigger(emote, mob/source)
+	if(emote == "deathgasp")
+		activate("death")
+
+/obj/item/weapon/implant/dust/activate(cause)
+	if(!cause || !imp_in || cause == "emp")
+		return 0
+	if(cause == "action_button" && alert(imp_in, "Are you sure you want to activate your dusting implant? This will turn you to ash!", "Dusting Confirmation", "Yes", "No") != "Yes")
+		return 0
+	to_chat(imp_in, "<span class='notice'>Your dusting implant activates!</span>")
+	imp_in.visible_message("<span class = 'warning'>[imp_in] burns up in a flash!</span>")
+	var/obj/effect/decal/remains/human/R = new /obj/effect/decal/remains/human(imp_in.loc)
```

Comment:
this should use `imp_in.dust()` instead. Then it'll call all the thing in `death` and such before spawning a skeleton and stuff.